src/server/kiwix-serve.cpp

```diff
@@ -384,12 +385,14 @@ int main(int argc, char** argv)
     exit(1);
   }
 
-  std::string host = (server.getIpMode()==kiwix::IpMode::all || server.getIpMode()==kiwix::IpMode::ipv6)
-                     ? "[" + server.getAddress() + "]" : server.getAddress();
-
-  std::string url = "http://" + host + ":" + std::to_string(server.getPort()) + normalizeRootUrl(rootLocation);
-  std::cout << "The Kiwix server is running and can be accessed in the local network at: "
-            << url << std::endl;
+  std::string prefix = "http://";
+  kiwix::IpAddress addresses = server.getAddress();
+  std::string suffix = ":" + std::to_string(server.getPort()) + normalizeRootUrl(rootLocation);
+  std::cout << "The Kiwix server is running and can be accessed in the local network at: " << std::endl;
+  bool validMode4 = (ipMode == kiwix::IpMode::ipv4 || ipMode == kiwix::IpMode::all);  // Mode is valid to support ipv4
+  bool validMode6 = (ipMode == kiwix::IpMode::ipv6 || ipMode == kiwix::IpMode::all);  // Mode is valid to support ipv6
```

Comment:
I don't understand why you need these. I think that `Server::getAddress()` must return only the addresses actually in use by the server. If it doesn't work that way - fix it in `libkiwix`.